### PR TITLE
ci: Add anaconda pruning to CI pipeline

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44651 ci: Add anaconda pruning to CI pipeline**

Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23692851](https://our.internmc.facebook.com/intern/diff/D23692851)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1113,6 +1113,35 @@ jobs:
           cat "$script"
           source "$script"
 
+  anaconda_prune:
+    parameters:
+      packages:
+        type: string
+        description: "What packages are we pruning? (eg. pytorch, torchvision, torchaudio, etc.)"
```
Comment (walterddr) [inline .circleci/config.yml:1120]:
```suggestion
        description: "What packages are we pruning? (eg. pytorch, torchvision, torchaudio, etc.)"
        description: "What packages are we pruning? (quoted, space-separated string. eg. "pytorch", "torchvision torchaudio", etc.)"
```

File: .circleci/verbatim-sources/job-specs/binary-job-specs.yml [hunk 2]
```diff
@@ -324,3 +324,32 @@
...
+      - run:
+          name: Install dependencies
+          no_output_timeout: "1h"
+          command: |
+            conda install -yq anaconda-client
+      - run:
+          name: Prune packages
+          no_output_timeout: "1h"
+          command: |
+              ANACONDA_API_TOKEN="${CONDA_PYTORCHBOT_TOKEN}" \
```
Comment (walterddr) [inline .circleci/verbatim-sources/job-specs/binary-job-specs.yml:353]: this might be a dumb question but where is `CONDA_PYTORCHBOT_TOKEN` set?

**Review by walterddr — APPROVED** [review]
lgtm

**Review by peterjc123 — APPROVED** [review]

**Review by guyang3532 — APPROVED** [review]

File: .circleci/verbatim-sources/job-specs/binary-job-specs.yml (hunk 2 above)
Comment (seemethere) [reply .circleci/verbatim-sources/job-specs/binary-job-specs.yml:353]: Not a dumb question.

This is set within circleci's credential store under the `org-member` context.

**Pull request closed without merging**